src/colorizer/utils/data_utils.ts

```diff
@@ -327,3 +328,47 @@ export function getGlobalIdFromSegId(
 
   return rawGlobalId - 1; // -1 to convert to zero-based index
 }
+
+export function getLabelTypeFromParsedCsv(
+  headers: string[],
+  data: Record<string, string | undefined>[]
+): Map<string, LabelType> {
+  const labelTypeMap = new Map<string, LabelType>();
+  for (const header of headers) {
+    let hasIntegerValues = true;
+    let hasBooleanValues = true;
+    for (const row of data) {
+      const value = row[header]?.trim();
+      const valueAsInt = parseInt(value ?? "", 10);
+      if (value === undefined || value === "") {
+        continue;
+      } else if (value.toLowerCase() === BOOLEAN_VALUE_TRUE || value.toLowerCase() === BOOLEAN_VALUE_FALSE) {
```

Comment:
nit: the second `if` doesn't need to `else` off the first, and without the `else` the first `if` can go before the `parseInt` call and skip it for undefined values